commands/auth_utils.go

```diff
@@ -29,25 +37,41 @@ type Credentials struct {
 
 type CredentialsList map[string]*Credentials
 
-func getConfigFilePath() (string, error) {
-	user, err := user.Current()
+func getDefaultConfigPath() string {
+	return filepath.Join("$HOME", xdgConfigHome, "mmctl")
+}
+
+func resolveConfigFilePath() (string, error) {
+	u, err := user.Current()
 	if err != nil {
 		return "", err
 	}
-	return user.HomeDir + "/.mmctl", nil
+
+	configPath := strings.Replace(viper.GetString("config"), "$HOME", u.HomeDir, 1)
+
+	// probe user-defined or xdg-compliant default config path
+	if _, err := os.Stat(filepath.Join(configPath, configFileName)); err != nil {
+		if !os.IsNotExist(err) {
+			return "", err
+		}
+		// conditionally fall back to legacy config path
+		configPath = u.HomeDir
```

Comment:
Probably we want to show a warning or message if that happens giving feedback to the user

Reply:
Sure!

Reply:
> Regarding the tests, that would be awesome. You can just define the flag and clean up the files after that but probably Miguel could help you better

@mgdelacroix Do you have a good solution for how to test this? I'd like to avoid creating and deleting real files in the user's actual home directory. It would be fine for CI/CD, but if tests are run in a developers' machine, her local config might potentially be overwritten / deleted in the worst case.

Reply:
I think that the best option here is to use something like:

```go
var currentUser *user.User

func SetUser(newUser *user.User) {
	currentUser := newUser
}

func init() {
	newUser, err := user.Current()
	if err != nil {
		panic(err)
	}

	SetUser(newUser)
}
```

This way you can use `SetUser` at the beginning of the tests to mock a fake `HomeDir` like this:

```go
dir, _ := ioutil.TempDir("", "mmctl-")
defer os.RemoveAll(dir)

u, err := user.Current()
if err != nil {
	panic(err)
}

oldUser = currentUser
u.HomeDir = dir
SetUser(u)
defer func () {
    SetUser(oldUser)
}
```